python/pyiceberg/table/partitioning.py

```diff
@@ -157,3 +159,20 @@ def compatible_with(self, other: "PartitionSpec") -> bool:
 
 
 UNPARTITIONED_PARTITION_SPEC = PartitionSpec(spec_id=0)
+
+
+def assign_fresh_partition_spec_ids(spec: PartitionSpec, schema: Schema) -> PartitionSpec:
+    partition_fields = []
+    for pos, field in enumerate(spec.fields):
+        schema_field = schema.find_field(field.name)
```

Comment:
This is the partition field name, not a schema field name. The schema field must be looked up by `source_id`.

Reply:
Great catch! 👍🏻

Reply:
@Fokko, looks like this hasn't been fixed yet, so I'm reopening the thread.

Reply:
Sorry, that slipped through somehow